### PR TITLE
Update the screen-capture IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Replacement for the totally borken https://github.com/web-platform-tests/wpt/pull/11835

Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the interfaces/ directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by lukebjerring** [comment]
@foolip - https://github.com/web-platform-tests/wpt/pull/11835 is locked in a bad state after I pushed too much merge history by mistake. Though the branch underneath the PR matches this PR, the GitHub UI is locked up (maybe because of the quantity of labels).

File: screen-capture/idlharness.window.js
Comment (foolip) [inline screen-capture/idlharness.window.js]: Well, this is one of the more annoying discrepancies between TR and github.io I've seen :)

File: interfaces/screen-capture.idl
Comment (foolip) [inline interfaces/screen-capture.idl]: Yes, it's a bid odd that these enums aren't used. They're referenced in prose, similar to what you can see if searching for "VideoFacingModeEnum" in https://w3c.github.io/mediacapture-main/#dom-videofacingmodeenum

File: screen-capture/META.yml
Comment (foolip) [inline screen-capture/META.yml]: @martinthomson are you OK with getting reviews in this directory. @alvestrand I suspect it might make sense for you to keep an eye on this, can I add you too?

**Review by foolip — APPROVED** [review]
LGTM % suggested reviewers confirming.

File: interfaces/screen-capture.idl
Comment (martinthomson) [inline interfaces/screen-capture.idl]: So no reformatting on ingest?

**Review by martinthomson — APPROVED** [review]

File: interfaces/screen-capture.idl
Comment (foolip) [reply interfaces/screen-capture.idl]: Yep. It's harmless since defining an IDL enum doesn't do anything unless used as an attribute/argument type somewhere, but could come up as noise if we in the future build tooling to detect IDL "dead code". I looked at the spec and think that using prose do define the equivalent behavior would be the fix if later needed.

File: screen-capture/META.yml
Comment (foolip) [reply screen-capture/META.yml]: @suhasHere, I've sent an invite to the web-platform-tests org. You'll need to accept that or it won't be possible for wpt-pr-bot to assign you as a reviewer to new PRs, so please accept the invite at the top of https://github.com/web-platform-tests.

File: interfaces/screen-capture.idl
Comment (foolip) [reply interfaces/screen-capture.idl]: No, lots of these IDL files look a bit funny as a result. This would be a feature request on https://github.com/tidoust/reffy/issues, IDL is imported verbatim from https://github.com/tidoust/reffy-reports.

File: screen-capture/META.yml
Comment (foolip) [inline screen-capture/META.yml]: @alvestrand confirmed to me on chat this is OK.

File: screen-capture/META.yml
Comment (foolip) [inline screen-capture/META.yml]: @suhasHere, can you confirm that you're OK with getting reviews for this test suite, and that you've accepted the org invite? Thanks!

File: screen-capture/META.yml
Comment (foolip) [inline screen-capture/META.yml]: @uysalere has also confirmed on chat

**Comment by foolip** [comment]
In the interest of getting this merged, I'll go ahead and merge this and send a separate PR for @suhasHere which can be merged when he has joined the org.

**Pull request merged**
